src/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,274 @@
+  /** bouncinessOut animation when press */
+  bouncinessOut?: number
+  /** define custom icon component */
+  iconComponent?: React.ReactNode
+  /** define custom text component */
+  textComponent?: React.ReactNode
+  /** custom style for the icon */
+  iconStyle?: CustomStyleProp
+  /** custom style for the inner icon */
+  innerIconStyle?: CustomStyleProp
+  /** style of container view  */
+  style?: CustomStyleProp
+  /** style of checkbox text */
+  textStyle?: CustomTextStyleProp
+  /** style icon image */
+  iconImageStyle?: CustomImageStyleProp
+  /** checkbox text container style */
+  textContainerStyle?: CustomStyleProp
+  /** define image source show when checkbox is checked */
+  checkIconImageSource?: ImageSourcePropType
+  /** callback when checkbox is change state */
+  onPress?: (checked: boolean) => void
+}
+
+interface ICheckboxMethods {
+  onHandlePress: () => void
+}
+
+const defaultCheckImage = require('../assets/images/check.png')
+
+const Checkbox = React.forwardRef<ICheckboxMethods, ICheckboxProps>(
+  (
+    {
+      style,
+      size = 25,
+      iconStyle,
+      iconComponent,
+      iconImageStyle,
+      fillColor = '#ffc484',
+      unfillColor = 'transparent',
+      disableBuiltInState = false,
+      isChecked,
+      innerIconStyle,
+      checkIconImageSource = defaultCheckImage,
+      text,
+      textComponent,
+      textStyle,
+      textContainerStyle,
+      disableText = false,
+      disable = false,
+      disableOpacity = 0.5,
+      bounceEffectIn = 0.9,
+      bounceEffectOut = 1,
+      bounceVelocityIn = 0.1,
+      bounceVelocityOut = 0.4,
+      bouncinessIn = 20,
+      bouncinessOut = 20,
+      useNativeDriver = true,
+      onPress,
+      ...rest
+    },
+    forwardedRef,
+  ) => {
+    const [checked, setChecked] = useState(false)
+    const bounceValue = useRef(new Animated.Value(1)).current
+
+    useEffect(() => {
+      setChecked(isChecked ?? false)
+    }, [isChecked])
+
+    const bounceInEffect = () => {
+      Animated.spring(bounceValue, {
+        toValue: bounceEffectIn,
+        velocity: bounceVelocityIn,
+        bounciness: bouncinessIn,
+        useNativeDriver,
+      }).start()
+    }
+
+    const bounceOutEffect = () => {
+      Animated.spring(bounceValue, {
+        toValue: bounceEffectOut,
+        velocity: bounceVelocityOut,
+        bounciness: bouncinessOut,
+        useNativeDriver,
+      }).start()
+    }
+
+    const syntheticBounceEffect = () => {
+      Animated.sequence([
+        Animated.timing(bounceValue, {
+          toValue: bounceEffectIn,
+          duration: 50,
+          useNativeDriver,
+        }),
+        Animated.spring(bounceValue, {
+          toValue: bounceEffectOut,
+          velocity: bounceVelocityOut,
+          bounciness: bouncinessOut,
+          useNativeDriver,
+        }),
+      ]).start()
+    }
+
+    const renderCheckIcon = () => {
+      const checkStatus = disableBuiltInState ? isChecked! : checked
```

Comment:
isChecked cannot be null or undefined ?

Reply:
you mean

Reply:
write "isChecked!" you are telling to the compiler that isCheck variable cannot be null or undefined.